### ISSUE TITLE
Test output for toBe in the case of reference comparison

### ISSUE THREAD

**Issue by DuncanUszkay1** [jtenner/as-pect#266]
If I have this class:
```typescript
export class Foo {
  someField: i32
}
```
And this test:
```typescript
describe("When a reference check is used", () => {
   it("should fail even when fields are the same", () => {
     let a = new Foo()
     let b = new Foo()
     a.someField = 1
     b.someField = 1
     expect<Foo>(a).toBe(b)
   });
 })
```
It fails as expected. Because Foo doesn't have the == operator overridden reference equality is used, so the fact that the fields match is irrelevant. What I think could be improved is the error output in this case- here's what you get now:
```
❌ discount.spec.ts Pass: 14 / 15 Todo: 0 Time: 16.313ms
  Failed: When a reference check is used
    ❌ should fail even when fields are the same -
      [Expected]: Foo {
      someField: 1
  }
      [Actual]  : Foo {
      someField: 1
  }
```
With this output, the difference between the expected and the actual (the reference) is not visible.

I'm not sure what the best way forward here is- maybe the reference could be included in the output:
```
❌ discount.spec.ts Pass: 14 / 15 Todo: 0 Time: 16.313ms
  Failed: When a reference check is used
    ❌ should fail even when fields are the same -
      [Expected]: Foo(0xdeadbeef) {
      someField: 1
  }
      [Actual]  : Foo(0xdeaddead) {
      someField: 1
  }
```
Or maybe it should revert to the old pre-reflection functionality in the case of a reference comparison:
```
❌ discount.spec.ts Pass: 14 / 15 Todo: 0 Time: 16.313ms
  Failed: When a reference check is used
    ❌ should fail even when fields are the same -
      [Expected]: Reference Value
      [Actual]  : Reference Value
  }
```
Thoughts?

**Comment by jtenner** [comment]
You can always improve your error output by providing a string that explains your assertion.

```ts
expect(a).toBe(b,  "A and B should be exactly the same reference");
```

Please see the assembly package for examples on how to write these messages.

**Comment by jtenner** [comment]
fixed in #268
